src/asm/output.c

```diff
@@ -532,13 +549,25 @@ static void checksectionoverflow(uint32_t delta_size)
  */
 void out_WriteObject(void)
 {
+	/* Write to a temporary file in the target's folder */
+	char *objectNameCopy = strdup(tzObjectname);
+	char const *dirPath = dirname(objectNameCopy);
+	char tmpFileName[strlen(dirPath) + 1 + 16 + 1];
+
+	sprintf(tmpFileName, "%s/rgbasm_tmpXXXXXX", dirPath);
```

Comment:
``snprintf``?

Reply:
Guaranteed not to overflow given the buffer's size.

Reply:
Technically, given that the buffer is a VLA, it's not even guaranteed to exist.

Reply:
Even if it was correct, I think that adding numbers by hand makes it quite fragile, so it's better to be defensive and use ``snprintf`` unless there is a really good reason not to do it.